src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -140,21 +142,23 @@ class LibraryService (protected val argUserInfo: UserInfo,
     }
   }
 
+
+  // should only be used to change published state
   def setWorkspaceIsPublished(ns: String, name: String, value: Boolean): Future[PerRequestMessage] = {
     rawlsDAO.getWorkspace(ns, name) flatMap { workspaceResponse =>
       val pub = isPublished(workspaceResponse)
-      val requiredLevel = if (!pub || (pub && !value)) WorkspaceAccessLevels.Owner else WorkspaceAccessLevels.Write
-      if (workspaceResponse.accessLevel >= requiredLevel) {
-        val operations = updatePublishAttribute(value)
-        rawlsDAO.patchWorkspaceAttributes(ns, name, operations) map { ws =>
-          if (value)
-            publishDocument(ws)
-          else
-            removeDocument(ws)
-          RequestComplete(ws)
+      if (pub == value)
+        Future(RequestCompleteWithErrorReport(NotModified, s"No changes needed"))
```

Comment:
I don't like using 304 here ... 304 is not an error and has implications for client caching. How about returning `Future(RequestComplete(NoContent))` here?